src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/incentives/PrisonerFromNomisIntTest.kt

```diff
@@ -18,46 +23,85 @@ class PrisonerFromNomisIntTest : SqsIntegrationTestBase() {
+
+      await untilAsserted { mappingApi.verifyCreateIncentiveMapping() }
+
+      verify(telemetryClient).trackEvent(
+        eq("incentive-created-synchronisation"),
+        eq(
+          mapOf(
+            "bookingId" to "1234",
+            "incentiveSequence" to "1",
+            "incentiveId" to "654321",
+            "auditModuleName" to "OIDOIEPS"
+          )
+        ),
+        isNull()
+      )
+    }
   }
 
-  @Test
-  fun `will consume a prison offender events message for an NOMIS updated incentive`() {
-
-    val message = validIepCreatedMessage()
-
-    nomisApi.stubGetIncentive(bookingId = 1234, incentiveSequence = 1)
-    mappingApi.stubIncentiveMappingByNomisIds(nomisBookingId = 1234, nomisIncentiveSequence = 1)
-
-    awsSqsOffenderEventsClient.sendMessage(queueOffenderEventsUrl, message)
-
-    await untilAsserted { nomisApi.verifyGetIncentive(bookingId = 1234, incentiveSequence = 1) }
-
-    verify(telemetryClient).trackEvent(
-      eq("incentive-updated-synchronisation"),
-      eq(mapOf("bookingId" to "1234", "incentiveSequence" to "1", "auditModuleName" to "OIDOIEPS")),
-      isNull()
-    )
-
-    // TODO - add incentive api update call
+  @Nested
+  @DisplayName("synchronise update incentive")
+  inner class SynchroniseUpdate {
+    @Test
+    fun `will synchronise an incentive after a nomis update to a current incentive`() {
+
+      val message = validIepCreatedMessage()
+
+      nomisApi.stubGetIncentive(bookingId = 1234, incentiveSequence = 1, currentIep = true)
+      mappingApi.stubIncentiveMappingByNomisIds(nomisBookingId = 1234, nomisIncentiveSequence = 1)
+      incentivesApi.stubUpdateSynchroniseIncentive()
+      awsSqsOffenderEventsClient.sendMessage(queueOffenderEventsUrl, message)
+
+      await untilAsserted { incentivesApi.verifyUpdateSynchroniseIncentive(1) }
+
+      verify(telemetryClient, Times(1)).trackEvent(
+        eq("incentive-updated-synchronisation"),
+        eq(
+          mapOf(
+            "bookingId" to "1234",
+            "incentiveSequence" to "1",
+            "auditModuleName" to "OIDOIEPS",
+            "currentIep" to "true"
+          )
+        ),
+        isNull()
+      )
+    }
+
+    @Test
+    fun `will synchronise an incentive after a nomis update to a non-current incentive`() {
+
+      val message = validIepCreatedMessage()
+
+      nomisApi.stubGetIncentive(bookingId = 1234, incentiveSequence = 1, currentIep = false)
+      mappingApi.stubIncentiveMappingByNomisIds(nomisBookingId = 1234, nomisIncentiveSequence = 1)
+      incentivesApi.stubUpdateSynchroniseIncentive()
+      awsSqsOffenderEventsClient.sendMessage(queueOffenderEventsUrl, message)
+      nomisApi.stubGetCurrentIncentive(bookingId = 1234, incentiveSequence = 2)
+
+      await untilAsserted { incentivesApi.verifyUpdateSynchroniseIncentive(2) }
+
+      verify(telemetryClient, Times(2)).trackEvent(
+        eq("incentive-updated-synchronisation"),
+        any(),
+        isNull()
+      )
```

Comment:
might need to switch to MocKK to verify the 2 different app insights event bodies